SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Ingestion/Http/IngestionHttp.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Azure.Mobile.Ingestion.Models;
+using Microsoft.Azure.Mobile.Utils;
+
+namespace Microsoft.Azure.Mobile.Ingestion.Http
+{
+   public sealed class IngestionHttp : IIngestion
+    {
+        internal const string DefaultBaseUrl = "https://in.mobile.azure.com";
+        internal const string ApiVersion = "/logs?api_version=1.0.0-preview20160914";
+        internal const string ContentTypeValue = "application/json; charset=utf-8";
+        internal const string AppSecret = "App-Secret";
+        internal const string InstallId = "Install-ID";
+
+        private readonly TimeSpan _requestTimeout = TimeSpan.FromMilliseconds(80000); //TODO not sure what to use here
+        private const int MaximumCharactersDisplayedForAppSecret = 8;
+        private string _baseLogUrl;
+        private readonly IHttpNetworkAdapter _httpNetwork;
+
+        public IngestionHttp() : this(new HttpNetworkAdapter())
+        {
+        }
+
+        public IngestionHttp(IHttpNetworkAdapter httpNetwork)
+        {
+            _httpNetwork = httpNetwork;
+            _httpNetwork.Timeout = _requestTimeout;
+        }
+
+        /// <exception cref="IngestionException"/>
+        public async Task ExecuteCallAsync(IServiceCall call)
+        {
+            if (call.CancellationToken.IsCancellationRequested)
+            {
+                return;
+            }
+            var requestContent = CreateLogsContent(call.Logs);
+            var request = CreateRequest(call.AppSecret, call.InstallId, requestContent);
+            HttpResponseMessage response;
+            try
+            {
+                response = await _httpNetwork.SendAsync(request, call.CancellationToken).ConfigureAwait(false);
```

Comment:
you might want to refactor this into a using. If someone edits this later they may miss a code path where request is not disposed.